src/s3/explorer/s3Nodes.ts

```diff
@@ -12,24 +12,25 @@ import { PlaceholderNode } from '../../shared/treeview/nodes/placeholderNode'
 import { makeChildrenNodes } from '../../shared/treeview/treeNodeUtilities'
 import { S3BucketNode } from './s3BucketNode'
 import { inspect } from 'util'
+import { AWSCommandTreeNode } from '../../shared/treeview/nodes/awsCommandTreeNode'
 
 /**
  * An AWS Explorer node representing S3.
  *
  * Contains buckets for a specific region as child nodes.
  */
 export class S3Node extends AWSTreeNodeBase {
-    public constructor(private readonly s3: S3Client) {
+    private readonly addedBuckets: S3BucketNode[] = []
```

Comment:
What happens if you add the same bucket more than once?

Reply:
You'll have the same showing twice, obviously :)
(This should be a map, not an array)